src/components/CountrySelector/CountrySelector.tsx

```diff
@@ -49,21 +44,6 @@ export default function CountrySelector<
 
   const [query, setQuery] = useState(country.name);
 
-  /**
```

Comment:
removing this `useEffect`  as it makes the form it belongs to `dirty` (related to #2467 ), Forms using `<CountrySelector/>` should now take care of setting flags for persistent country records
see changes: 
EditProfile
Documentation